Python/Product/PythonTools/Snippets/Python/argparse.snippet

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<CodeSnippets xmlns="http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet">
+  <CodeSnippet Format="1.0.0">
+    <Header>
+      <Title>argparse block</Title>
+      <Shortcut>argparse</Shortcut>
+      <Description>Code snippet for a function to parse arguments using argparse</Description>
+      <Author>Microsoft Corporation</Author>
+      <SnippetTypes>
+        <SnippetType>Expansion</SnippetType>
+      </SnippetTypes>
+    </Header>
+    <Snippet>
+      <Imports>
+        <Import>
+          <Namespace>argparse</Namespace>
+        </Import>
+        <Import>
+          <Namespace>sys</Namespace>
+        </Import>
+      </Imports>
+      <Declarations>
+        <Literal>
+          <ID>description</ID>
+          <ToolTip>Short description of your script</ToolTip>
+          <Default>My Python script.</Default>
+        </Literal>
+      </Declarations>
+      <Code Language="Python"><![CDATA[def parse_args(argv=sys.argv[1:]):
+    parser = argparse.ArgumentParser(description="$description$")
+
+    # Use the arg_* snippets to add arguments
+    $end$
```

Comment:
What does $end$ do and why is it not at the end in this one?

Reply:
When you don't support surrounding a selection, `$end$` is where the caret should be put when you commit the snippet.